<launch>
    <node pkg="map_server" type="map_server" name="map_server" args="$(find lidar_simulation)/map/custom.yaml" />

    <node pkg="lidar_simulation" type="lidar_simulation" name="lidar" output="screen">
        <param name="min_angle"  value="-3.141592653"/>
        <param name="max_angle"  value="3.141592653" />
        <param name="min_distance" value="0.15" />
        <param name="max_distance" value="6.00" />
        <param name="size" value="400"/>
        <param name="rate" value="10"/>
    </node>


    <node pkg="autolabor_fake" type="autolabor_fake_node" name="autolabor_driver" >
        <remap from="/autolabor/cmd_vel" to="cmd_vel" />
        <remap from="/autolabor/odom" to="odom" />
    </node>

    <node name="obstacle_simulation" pkg="lidar_simulation" type="obstacle_simulation" output="screen">
	<param name="obstacle_id" value="obstacle" />
        <param name="inter_marker_id" value="map" />
        <param name="resolution" value="0.05" />
        <!--<rosparam param="obstacle_footprints">
        [[0.1,1.0, 0.1,-1.0, -0.1,-1.0, -0.1,1.0]]
        </rosparam>-->
        <rosparam param="obstacle_footprints">
        [[0.5,0.0, 0.1,0.1, 0.0,0.5, -0.5,0.0, 0.0,-0.5, 0.1,-0.1]]
        </rosparam>
    </node>

    <node name="obstacle_simulation_1" pkg="lidar_simulation" type="obstacle_simulation" output="screen">
        <param name="obstacle_id" value="other" />
        <param name="inter_marker_id" value="map" />
        <param name="resolution" value="0.05" />
        <rosparam param="obstacle_footprints">
        [[0.1,1.0, 0.1,-1.0, -0.1,-1.0, -0.1,1.0]]
        </rosparam>
<!--        <rosparam param="obstacle_footprints">
        [[0.5,0.0, 0.1,0.1, 0.0,0.5, -0.5,0.0, 0.0,-0.5, 0.1,-0.1]]
        </rosparam>-->
    </node>

    <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0.0 0.0 0.0 0.0 0.0 0.0 /map /odom 10" />
    <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser" args="0.0 0.0 0.20 0.0 0.0 0.0 /base_link /lidar 10" />
</launch>
